### PR TITLE
use data-css-strict on amp4email

### LINKED ISSUE #5053: Add data-css-strict to all email samples
This enables stricter CSS validation required for Gmail compatibility.

//cc @patrickkettner

### PR DESCRIPTION
**Pull request by patrickkettner**
fixes #5053

### CODE REVIEW

File: examples/source/1.components/amp-autocomplete.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
 -->
 <!-- -->
 <!doctype html>
-<html ⚡ lang="en">
+<html ⚡ lang="en" data-css-strict>
```
Comment (sebastianbenz) [inline examples/source/1.components/amp-autocomplete.html:11]: This is not an email only sample, we should put this behind a template variable that will only inject the flag if the email mode is enabled.

**Review by sebastianbenz — CHANGES_REQUESTED** [review]
Thanks for the quick fix!

File: examples/source/1.components/amp-autocomplete.html (hunk 1 above)
Comment (patrickkettner) [reply examples/source/1.components/amp-autocomplete.html:11]: this was just a mistake - email has [it's own autocomplete example](https://github.com/ampproject/amp.dev/blob/use-data-css-strict/examples/source/1.components/amp-autocomplete-email.html)

Comment (sebastianbenz) [reply examples/source/1.components/amp-autocomplete.html]: Afaik, some examples are rewritten for email.

Comment (patrickkettner) [reply examples/source/1.components/amp-autocomplete.html]: I believe `amp-bind-macro` is the only example of that, I will get that one sorted shortly

Comment (sebastianbenz) [reply examples/source/1.components/amp-autocomplete.html]: Found it, we need to add the flag here as well: https://github.com/ampproject/amp.dev/blob/future/platform/lib/format-transform/formats/amp4email.js#L26

Comment (patrickkettner) [reply examples/source/1.components/amp-autocomplete.html]: beauty, thanks. I would've taken me a long while to find that.

pushed and update for that and for the test suite

**Review by sebastianbenz — APPROVED** [review]
Thanks!

**Pull request merged**